pgxn/neon/walproposer_pg.c

```diff
@@ -272,6 +272,30 @@ split_safekeepers_list(char *safekeepers_list, char *safekeepers[])
 	return n_safekeepers;
 }
 
+static char *split_off_safekeepers_generation(char *safekeepers_list, uint32 *generation)
+{
+	char	   *endptr;
+
+	if (strncmp(safekeepers_list, "g#", 2) != 0)
```

Comment:
it can be done simpler using sscanf:
```
int pos;
if (sscanf(safekeepers_list, "g#%u:%n", generation, &pos) == 1) { 
     return safekeepers_list + pos;
} else {
     return safekeepers_list;
}
```